Tasks/AzureResourceGroupDeploymentV2/models/DeployAzureRG.ts

```diff
@@ -99,13 +99,21 @@ export class AzureRGTaskParameters {
     
     private _getAzureADGraphCredentials(connectedService: string): msRestAzure.ApplicationTokenCredentials {
         var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
-        var servicePrincipalKey: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalkey", false);
+        var servicePrincipalAuth: string;
+        let authType: string = tl.getEndpointAuthorizationParameter(connectedService, 'authenticationType', true);
```

Comment:
I think we also need to check for servicePrincipal auth scheme otherwise managed identity will not work

Reply:
Also, can managed identity be used to get graph token? Can we test it?

Reply:
Managed identity is not AAD Application, it's a way to manage resources without Service Principal. Since Graph deals with AAD Apps, we cannot get graph token using Managed Identity. I'll update to check for auth scheme. Thanks a lot!

Reply:
The Service Principal used by Managed Identity is managed by Azure, so we don't have the secret used to create MI. Graph API requires us to give the client secret to fetch Auth token, so it's not possible.

I'll update to check auth scheme. Thanks a lot!

Reply:
I thought so, but not so sure. Is there a doc which explains this fact? Basically my doubt is that MI itself is an identity which can make request to ARM, so why can't it make request to Graph.

Reply:
After further investigation, I found that it can fetch graph token, if the fetching is done from resource. So yes VSTS Agent using MI might be able to fetch, though not using Graph API. It can do so using the REST API which we use to fetch ARM Token for MI.